File: backend/src/main/java/com/nexus/backend/service/AuthService.java
package com.nexus.backend.service;

import com.nexus.backend.dto.request.LoginRequest;
import com.nexus.backend.dto.request.RegisterRequest;
import com.nexus.backend.dto.response.AuthResponse;

public interface AuthService {

    // Existing
    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    // OTP Registration
    String sendRegistrationOtp(String email);

    String verifyRegistrationOtpCode(
            String email,
            String otp
    );

    AuthResponse verifyRegistrationOtp(RegisterRequest request);

    // Forgot Password
    String sendForgotPasswordOtp(String email);

    String resetPassword(
            String email,
            String otp,
            String newPassword,
            String confirmPassword
    );
}

File: frontend/src/services/authService.js
import api from "./api";

// ==========================================
// LOGIN
// ==========================================
export const loginUser = async (email, password) => {
  const response = await api.post("/auth/login", {
    email,
    password,
  });

  return response.data;
};

// ==========================================
// REGISTER
// ==========================================
export const registerUser = async (userData) => {
  const response = await api.post("/auth/register", userData);

  return response.data;
};

// ==========================================
// SEND REGISTRATION OTP
// ==========================================
export const sendRegistrationOtp = async (data) => {
  const response = await api.post(
      "/auth/send-registration-otp",
      data
  );

  return response.data;
};

// ==========================================
// VERIFY REGISTRATION OTP
// ==========================================
export const verifyRegistrationOtp = async (data) => {
  const response = await api.post(
      "/auth/verify-registration-otp",
      data
  );

  return response.data;
};

// ==========================================
// SEND FORGOT PASSWORD OTP
// ==========================================
export const sendForgotPasswordOtp = async (data) => {
  const response = await api.post(
      "/auth/send-forgot-password-otp",
      data
  );

  return response.data;
};

// ==========================================
// RESET PASSWORD
// ==========================================
export const resetPassword = async (data) => {
  const response = await api.post(
      "/auth/reset-password",
      data
  );

  return response.data;
};